vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java

```diff
@@ -440,6 +465,11 @@ private AssertionCollection getAssertionCollection()
         return testContext.get(AssertionCollection.class, AssertionCollection::new);
     }
 
+    private FailTestFastManager getFailTestFastManager()
+    {
+        return testContext.get(FailTestFastManager.class, failTestFastManagerFactory::createFailTestFastManager);
```

Comment:
why is it needed to create own manager for each thread?

Reply:
If we setup two or more cores for test run we accordingly can have two or more stories which are running parallel. And if story with resource validation changing singleton manager, other stories can "lose" correct predefined fail-fast value